### PR TITLE
Candidate profile

### PR DESCRIPTION
**Pull request by Safi-Ullah**
## Issue:
https://github.com/fenske/devskills-app/issues/229

## What New: 
- Separate page for `Sent Assignment`.
- Each candidate's page has its' own unique URL (/candidates/{candidate_id} where candidate_id is an external candidate's ID of type UUIDv4).
- Backfill the new id field across all docs under the assignments collections.

## Additional Info:
- Separate script to populate `ids` for `assignments` collection.
- The code for `CandidateOverviewSlideover` & `UpdateArchived` endpoint is also still present even though it's not being used.

### CODE REVIEW

**Review by fenske — CHANGES_REQUESTED** [review]
Hi Safi. FYI, I initialized the id's on the preview env. 

I opened it on a 13-inch screen and it rendered really weird. 

Let me share a few comments:

CleanShot 2021-06-25 at 07 08 15@2x

I might have missed some more details, but the idea is that the implementation should copy the mockup quite closely.

Let me know if you have any other questions.

**Comment by Safi-Ullah** [comment]
This is what it was supposed to look like. 

Screenshot 2021-06-25 at 11 21 03 AM

The deployment is looking really weird for me as well. I'll have a look what's causing this.

**Comment by fenske** [comment]
I see. It could be the "purge" util removing the css classes. 🤔

**Comment by Safi-Ullah** [comment]
> I see. It could be the "purge" util removing the css classes. 🤔

But shouldn't it only remove unused css classes..?

**Comment by fenske** [comment]
It should. But I actually experienced problems with purge on a diff repo recently as it was removing too many CSS classes.

In tailwind.config.css, try to update the purge config to this:
```
purge: ['./{components,pages,lib}/**/*.tsx'],
```

**Comment by fenske** [comment]
Ok, that was it. 👍

Looks better now :) 

CleanShot 2021-06-25 at 08 32 17@2x

**Comment by Safi-Ullah** [comment]
> Ok, that was it. 👍
> 
> Looks better now :)
> 
> CleanShot 2021-06-25 at 08 32 17@2x

Yeah, I still see 1, 2 issues pointed above...will commit those changes as well....

**Comment by fenske** [comment]
I'll be off for a bit. Will complete the review by eod. 

In the meantime, could you please update the repo link to how it is on the mockup?

**Comment by fenske** [comment]
Thanks Safi! Have a good day!

**Comment by Safi-Ullah** [comment]
> I'll be off for a bit. Will complete the review by eod.
> 
> In the meantime, could you please update the repo link to how it is on the mockup?

Yeah, sure...along with adding space between the slider containers.

File: components/CandidateInfoPanel.tsx [hunk 1]
```diff
@@ -1,41 +1,45 @@
...
-    <div className="border-t border-gray-200 px-4 py-5 sm:px-6">
-      <dl className="grid grid-cols-1 gap-x-4 gap-y-8 sm:grid-cols-2">
-        <div className="sm:col-span-1">
-          <dt className="text-sm font-medium text-gray-500">Profile</dt>
-          <dd className="mt-1 text-sm text-gray-900 capitalize">{profile}</dd>
-        </div>
-        <div className="sm:col-span-1">
-          <dt className="text-sm font-medium text-gray-500">Email address</dt>
-          <dd className="mt-1 text-sm text-gray-900">{email}</dd>
+  return (
```
Comment (fenske) [inline components/CandidateInfoPanel.tsx:7]: Mainly highlighted the links + reformated a bit.

File: components/CandidateOverviewSlideover.tsx [hunk 2]
```diff
@@ -1,357 +0,0 @@
-import Transition from "./Transition";
```
Comment (fenske) [inline components/CandidateOverviewSlideover.tsx:1]: Not used anymore, hence deleted.

File: components/CandidateLablesPanel.tsx [hunk 3]
```diff
@@ -1,43 +1,46 @@
...
-        </button>
       </div>
-    </div>
-  </div>;
+  );
 };
 
-export default React.memo(CandidateLablesPanel);
\ No newline at end of file
+export default React.memo(CandidateLabelsPanel);
```
Comment (fenske) [inline components/CandidateLablesPanel.tsx:48]: Btw, nice use of `React.memo`. 👍

File: components/candidateList/CandidateRow.tsx [hunk 4]
```diff
@@ -3,19 +3,19 @@ import CandidateName from "./CandidateName";
...
     <li className="relative pl-4 pr-6 py-5 hover:bg-gray-50 sm:py-6 sm:pl-6 lg:pl-8 xl:pl-6">
       <div className="flex items-start justify-between space-x-4">
         <div className="min-w-0 space-y-3">
-          <div className="flex items-center space-x-3" onClick={e => {
-            e.preventDefault();
-            window.location.href = `/candidates/${challenge.id}`;
-          }}>
-            <CandidateName challenge={challenge}/>
-          </div>
+          <Link href={`/candidates/${challenge.id}`}>
```
Comment (fenske) [inline components/candidateList/CandidateRow.tsx:14]: `window.location.href` does a server-side routing whereas `Link` routes client-side. The performance boost is worth it.

**Review by fenske — APPROVED** [review]
@Safi-Ullah I pushed a couple of changes as I was reviewing the code. If you're ok with it, I'm happy to merge this PR.

Great work overall! We'll be further improving this view soon to add the assessment results to it, etc.

**Comment by fenske** [comment]
Also, one comment about the commit messages. 

For future work, it'd really help if you prepend each commit message with `#  <issue_number>`, e.g. `#229 Implemented a change".

It's handy because it's easy to relate to what issues each commit was associated with when browsing the commit log.

**Comment by Safi-Ullah** [comment]
> For future work, it'd really help if you prepend each commit message with `# <issue_number>`, e.g. `#229 Implemented a change".

Sure, will do from now on.

**Comment by Safi-Ullah** [comment]
> @Safi-Ullah I pushed a couple of changes as I was reviewing the code. If you're ok with it, I'm happy to merge this PR.

Yeah, I'm fine with this, you can go ahead and merge the PR.

**Pull request merged**